### ISSUE TITLE
USA Pathfinder is very strong

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#690]
USA Pathfinder is very strong. It does require a Level 3 Promotion, but still it is an absolute game changer. It kills all Infantry and Hero's instantaneously. It can even shoot when loaded in Humvee and Combat Chinook with laser precision.

Very strong vs:
* China Infantry
* USA
* GLA

It costs just 600 money and builds in 10 seconds. Perhaps it can be a bit less powerful while still remaining very useful.

### Proposal 1
Increase unit build cost.

### Proposal 2
Increase unit build time.

### Proposal 3
Decrease rate of fire.

### Proposal 4
Decrease damage.

### Proposal 5
Decrease rate of fire when loaded in vehicle only.

### Proposal 6
Decrease damage when loaded in vehicle only.

### Proposal 7
Reveals itself while shooting on foot (same as Jarmen Kell).

### Proposal 8
Does not reveal enemy stealth units.

### Proposal 9
Has reduced enemy stealth detection range.

### ~~Proposal 10~~
Has to aim for 1 second before shooting. This decreases kills per second.
Implementation: https://github.com/commy2/zerohour/commit/48b06f0d096ba37a7fd48ce210a534524c37929f
Would make unit feel unresponsive.

### Proposal 11
Make Pathfinder no longer use a hitscan weapon and use a dummy projectile instead. Dummy projectile doesn't track targets, but the weapon has a secondary damage radius with lower damage. Therefore, if Pathfinder hits stationary target, it gets hit with full damage and dies and if Pathfinder's target is moving, it'll only be hit by secondary damage, and require multiple hits to die.

### Proposal 12
Make Hero infantry more resistant to Pathfinders. This could also give Jarmen Kell an edge in 1v1 situations.

### ~~Proposal 13~~
Require Strategy Center to build Pathfinder.

**Comment by ExiLe-ZH** [comment]
think you meant production time with either proposal 1 or 2

**Comment by ExiLe-ZH** [comment]
Additional proposals:
- Reveals when shooting (like Jarmen Kell).
- Does not reveal stealth (a Lotus for example, Jarmen Kell doesn't either).

**Comment by commy2** [comment]
### Proposal 3b

Instead of straight decreasing the ROF, have it have to aim for 1 second before shooting. This decreases kills / second.

Implementation: https://github.com/commy2/zerohour/commit/48b06f0d096ba37a7fd48ce210a534524c37929f

**Comment by xezon** [comment]
Added as Proposal 10.

**Comment by ExiLe-ZH** [comment]
Another proposal is to nerf the damage so it's just enough to 1 shot non vetted standard infantry. The pathfinder would then need to vet up itself to 1 shot vetted infantry or heroes (would improve the entire mid/late game balance vs Inf gen).

**Comment by xezon** [comment]
It should be possible to tweak it like that yes. That is Proposal 4.

**Comment by ZekeDlyoung** [comment]
I suggested this idea before for another mod:
- Make Pathfinder no longer use a hitscan weapon and use a dummy projectile instead
- Dummy projectile doesn't track targets, but the weapon has a secondary damage radius with lower damage

Therefore:
- If Pathfinder hits stationary target, it gets hit with full damage and dies
- If Pathfinder's target is moving, it'll only be hit by secondary damage, and require multiple hits to die

This makes Pathfinders still good at dealing with infantry, but players can still make their infantry run away and suffer less casualties, or attempt to rush the Pathfinder and kill it if it's out in the open

**Comment by xezon** [comment]
Added as Proposal 11

**Comment by ReLaX82** [comment]
Add moving option as Zeke replied and make Pathis more expensive.

I would also decrease damage level for not moving anf moving targets.

**Comment by ExiLe-ZH** [comment]
Zeke's idea indeed sounds pretty cool

**Comment by xezon** [comment]
Fire^Arm
> Pathfinder is non hero unit but is stealth while shooting.
> Jarmen is hero unit but not stealth while sniping.
> Burton is stealth while knifing.

Is Proposal 7.

**Comment by xezon** [comment]
Regarding Proposal 13, an interesting option could be to give this requirement only to stronger USA faction variants, like Air Force, but not weaker ones, like Super Weapon.

**Comment by ReLaX82** [comment]
I wouldnt seperate between faction this way Xezon.

Test first and essentially let them cost more for some factions.

**Comment by HeissesFeuer** [comment]
I feel like requiring a strategy center would change a lot of the gameplay, especially in USA mirrors where both players are hunted and only on a supply and barrack. These situations usually result in pretty epic games.
Proposals 1, 5, 7, 8 and 12 would probably work great together

**Comment by HeissesFeuer** [comment]
I don't know if this is doable but would it be possible to give pathfinders different fire rates depending on the distance to the target?
like for far away targets the pathfinder would need to concentrate and re-calibrate the optics to make the shot.

**Comment by xezon** [comment]
> I feel like requiring a strategy center would change a lot of the gameplay

I agree with this. Limiting access to this unit would be very noticeable and likely a detrimental experience for many.

> I don't know if this is doable but would it be possible to give pathfinders different fire rates depending on the distance to the target?

I am not aware about any stock weapon that does it.

**Comment by ExiLe-ZH** [comment]
Had some ideas which probably are too mod'ish but interesting to think about.

Like imagine two shooting modes:

1. Shooting from normal standing position; it's quick, but reveals itself (like Jarmen), misses moving targets (or less damage) and with slower ROF.

2. The second shooting position is prone, it can hit moving targets (full damage), remains stealth while shooting and shoots faster (all exactly like it already is in 1.04). The downside however is the mobility and deployment time, same like a Nuke Cannon.
